mongodbatlas/fw_resource_mongodbatlas_database_user_test.go

```diff
@@ -1,281 +1,38 @@
-	expectedID := fmt.Sprintf("%s-%s-%s", requestModel.ProjectID.ValueString(), requestModel.Username.ValueString(), requestModel.AuthDatabaseName.ValueString())
-	if responseModel.ID.ValueString() != expectedID {
-		t.Fatalf("expected %s, got %s", expectedID, responseModel.ID)
-	}
-
-	if requestModel.Password != responseModel.Password {
-		t.Fatalf("expected %s, got %s", requestModel.Password, responseModel.Password)
-	}
-
-	if requestModel.AuthDatabaseName != responseModel.AuthDatabaseName {
-		t.Fatalf("expected %s, got %s", requestModel.AuthDatabaseName, responseModel.AuthDatabaseName)
-	}
-
-	if requestModel.ProjectID != responseModel.ProjectID {
-		t.Fatalf("expected %s, got %s", requestModel.AuthDatabaseName, responseModel.AuthDatabaseName)
-	}
-}
-
-func newDatabaseUserRSMock(ctx context.Context) (fwresource.Resource, *fwresource.SchemaResponse, error) {
-	schemaRequest := fwresource.SchemaRequest{}
-	schemaResponse := &fwresource.SchemaResponse{}
-
-	// Instantiate the resource.Resource and call its Schema method
-	resource := NewDatabaseUserRSWithClient(&MongoDBClient{
-		Atlas: &matlas.Client{
-			DatabaseUsers: &MockDatabaseUsersServiceOp{},
-		},
-	})
-
-	resource.Schema(ctx, schemaRequest, schemaResponse)
-	if schemaResponse.Diagnostics.HasError() {
-		return nil, nil, fmt.Errorf("Schema method diagnostics: %+v", schemaResponse.Diagnostics)
-	}
-
-	return resource, schemaResponse, nil
-}
-
-func newReadRequestResponse(schemaResponse *fwresource.SchemaResponse) (*fwresource.ReadRequest, *fwresource.ReadResponse) {
-	return &fwresource.ReadRequest{
-			State: tfsdk.State{
-				Schema: schemaResponse.Schema,
-			},
-		}, &fwresource.ReadResponse{
-			State: tfsdk.State{
-				Schema: schemaResponse.Schema,
-			},
-		}
-}
-
-func newCreateRequestResponse(schemaResponse *fwresource.SchemaResponse) (*fwresource.CreateRequest, *fwresource.CreateResponse) {
-	return &fwresource.CreateRequest{
-			Plan: tfsdk.Plan{
-				Schema: schemaResponse.Schema,
-			},
-		}, &fwresource.CreateResponse{
-			State: tfsdk.State{
-				Schema: schemaResponse.Schema,
-			},
-		}
-}
-
-func newUpdateRequestResponse(schemaResponse *fwresource.SchemaResponse) (*fwresource.UpdateRequest, *fwresource.UpdateResponse) {
-	return &fwresource.UpdateRequest{
-			Plan: tfsdk.Plan{
-				Schema: schemaResponse.Schema,
-			},
-		}, &fwresource.UpdateResponse{
-			State: tfsdk.State{
-				Schema: schemaResponse.Schema,
-			},
-		}
-}
-
-func newDeleteRequestResponse(schemaResponse *fwresource.SchemaResponse) (*fwresource.DeleteRequest, *fwresource.DeleteResponse) {
-	return &fwresource.DeleteRequest{
-			State: tfsdk.State{
-				Schema: schemaResponse.Schema,
-			},
-		}, &fwresource.DeleteResponse{
-			State: tfsdk.State{
-				Schema: schemaResponse.Schema,
+func TestConfigRSDatabaseUser_basic(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		ProtoV6ProviderFactories: NewUnitTestProtoV6ProviderFactories(),
```

Comment:
Using the `NewUnitTestProtoV6ProviderFactories` and reusing the acceptance test framework